### PR TITLE
DSDEEPB-1563: View WDL and other method details when viewing a selected MC (from in a user workspace not method repo)

### PR DESCRIPTION
**Pull request by eddiebroad**
cleanup

cleanup and use EntityDetails for method details

call it as Method Details section

have rendertopbar call MethodDetailsViewer

### CODE REVIEW

**Comment by rushtong** [comment]
This branch needs a rebase.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs [hunk 1]
```diff
@@ -197,35 +215,42 @@
...
-        (config "name")]
-       [:div {:style {:display (when-not editing? "none")}}
-        (style/create-text-field {:ref "confname" :defaultValue (config "name")})])
-     (create-section-header "Inputs")
-     (input-output-list config "inputs" invalid-inputs editing?)
-     (create-section-header "Outputs")
-     (input-output-list config "outputs" invalid-outputs editing?)]))
+    [:div {}
+     [:div {:style {:marginLeft 330}}
+
```
Comment (rushtong) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: Remove blank line here.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs [hunk 2]
```diff
@@ -197,35 +215,42 @@
...
+         (clear-both)
+         [:div {:style {:marginRight "1em" :padding "0.5em" :marginBottom "0.5em"
+                        :backgroundColor (:exception-red style/colors)
+                        :display (if-not (contains? invalid-values key) "none")
+                        :width 450
+                        :border (str "1px solid " (:line-gray style/colors)) :borderRadius 2}}
+          (get invalid-values key)]]])]))
+
+
+
```
Comment (rushtong) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: Nitpick - remove a couple of these extra lines. Keep to one or two between methods.

**Comment by rushtong** [comment]
General comment - I don't like how the payload expand link bumps everything down. I would prefer that to be a modal. Not sure what others think, but having the "Launch Analysis" button get pushed down is jarring to the eye.

**Comment by rushtong** [comment]
Not directly related to this PR, but with all of the functionality (edit, delete, publish) in the left nav, it seems to be that Launch Analysis should be there too.
@dmohs - your feedback is also requested on this comment.

**Comment by rushtong** [comment]
@eddiebroad - Done with my review. In general, the code looks fine to me, as does the added display elements. I have two UI suggestions they aren't important enough to hold up this PR.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_configs/method_config_editor.cljs [hunk 3]
```diff
@@ -75,6 +79,38 @@
...
+                    (:namespace props)
+                    (:name props)
+                    (:snapshotId props))
+        :headers {"Content-Type" "application/json"}
+        :on-done (fn [{:keys [success? get-parsed-response status-text]}]
+                   (if success?
+                     (swap! state assoc :loaded-method (get-parsed-response))
+                     (swap! state assoc :error status-text)))}))})
+
+(defn- render-top-bar [config]
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_configs/method_config_editor.cljs]: The relationship between this and `render-top-bar-config` is not very clear to me. Can you do some renaming to add clarity?

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_configs/method_config_editor.cljs [hunk 4]
```diff
@@ -18,6 +19,8 @@
   (vec (remove blank? (map trim list))))
 
 
+
+
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_configs/method_config_editor.cljs]: These extra lines don't add value, in my opinion.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_configs/method_config_editor.cljs [hunk 5]
```diff
@@ -75,6 +79,38 @@
...
+                     (swap! state assoc :loaded-method (get-parsed-response))
+                     (swap! state assoc :error status-text)))}))})
+
+(defn- render-top-bar [config]
+  [MethodDetailsViewer
+   {:name (get-in config ["methodRepoMethod" "methodName"])
+    :namespace (get-in config ["methodRepoMethod" "methodNamespace"])
+    :snapshotId (get-in config ["methodRepoMethod" "methodVersion"])
+    :config config
+    }])
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_configs/method_config_editor.cljs]: Style.

**Comment by dmohs** [comment]
Just a few comments to address.

**Comment by dmohs** [comment]
:+1:

**Pull request merged**